File: app/src/main/java/de/hype/hypenotify/tools/bazaar/BazaarResponse.java
package de.hype.hypenotify.tools.bazaar;

import java.time.Duration;
import java.time.Instant;
import java.util.Map;

public class BazaarResponse {
    private boolean success;
    private Long lastUpdated;
    private Map<String, BazaarProduct> products;

    public Map<String, BazaarProduct> getProducts() {
        return products;
    }

    public Instant getLastUpdated() {
        return Instant.ofEpochSecond(lastUpdated);
    }

    public boolean isOlderThan(Duration maxAge){
        return Instant.now().getEpochSecond() - lastUpdated > maxAge.getSeconds();
    }
}

File: app/src/main/java/de/hype/hypenotify/app/tools/bazaar/BazaarResponse.kt
package de.hype.hypenotify.app.tools.bazaar

import java.time.Duration
import java.time.Instant

class BazaarResponse {
    private val success = false
    private val lastUpdated: Long? = null
    val products: MutableMap<String, BazaarProduct> = mutableMapOf()

    fun getLastUpdated(): Instant? {
        return if (lastUpdated == null) null else Instant.ofEpochMilli(lastUpdated)
    }

    fun isOlderThan(maxAge: Duration): Boolean {
        if (lastUpdated == null) return true
        return Instant.now().toEpochMilli() - lastUpdated > maxAge.seconds
    }
}